pkg/cmd/cli/cmd/generate.go

```diff
@@ -109,3 +96,24 @@ $ s2i generate docker.io/centos/nodejs-10-centos7 Dockerfile.gen
 
 	return generateCmd
 }
+
+// manageConfigImageLabelsBuildImageName returns an error "for possible future use", but so far
+// the callers so far log only log a warning and continue, so as to not "require" the presence
+// of docker with 's2i generate' and 's2i build --as-dockerfile'
+func manageConfigImageLabelsBuildImageName(ctx context.Context, cfg *api.Config) error {
```

Comment:
This needs a better godoc - I think a) we can error if the builder image pull spec is invalid, and b) probably is not necessary to say that the methods we call won't return an error. If their signature includes an error, we should assume that it can return an error.

```suggestion
// manageConfigImageLabelsBuildImageName extracts the image labels from builder image in the provided config.
// Returns an error if the builder image name is invalid or there is an error extracting the image labels.
func manageConfigImageLabelsBuildImageName(ctx context.Context, cfg *api.Config) error {
```